sql/core.go

```diff
@@ -151,9 +151,9 @@ type WindowAdaptableExpression interface {
 type WindowFramer interface {
 	// NewFramer is a prototype constructor that create a new Framer with pass-through
 	// parent arguments
-	NewFramer(WindowInterval) WindowFramer
+	NewFramer(WindowInterval) (WindowFramer, error)
```

Comment:
until we push validation checks into analyzer, the range expression is validated in framer constructor